cityofaustin/atd-moped
# 7777: New UI element for sync to Data Tracker

## Associated issues

This PR intends to close https://github.com/cityofaustin/atd-data-tech/issues/7777. 

## Testing
https://flh-7777-sync-ui-element--atd-moped-main.netlify.app/moped/projects

NB: The staging database has a number of projects which indicate they are synchronized to knack, but the development knack database has had the corresponding records removed in the course of development. I would be happy to get connected to the staging database and remove these stale knack IDs - please let me know if this would be helpful.

NB: Also, please note that the URL that is used when sending the user to knack after they click on the "View in Data Tracker" link is based on the production Knack URL even when being used in testing. I attempted to put in some logic to direct the user to the development Data Tracker instance, but the SSO mechanism employed by Knack prevents this. 

**Steps to test:**
* Find a project which does not have a knack_id associated with it yet. You'll know you've found one when you see "Synchronize" being displayed by this component. 
* Click the link to start a sync attempt.
* Observe app behavior & observe the record having been created in the development knack data tracker app.

---
#### Ship list
